doc/modules/neighbors.rst

```diff
@@ -427,6 +483,38 @@ and the ``'effective_metric_'`` is in the ``'VALID_METRICS'`` list of
 same order as the number of training points, and that ``leaf_size`` is 
 close to its default value of ``30``.
 
+Valid Metrics for Nearest Neighbor Algorithms
+---------------------------------------------
+
+========================     =================================================================
+Algorithm                    Valid Metrics
+========================     =================================================================
+**Brute Force**              'euclidean', 'l2', 'l1', 'manhattan', 'cityblock',
+                             'braycurtis', 'canberra', 'chebyshev', 'correlation',
+                             'cosine', 'dice', 'hamming', 'jaccard', 'kulsinski',
+                             'mahalanobis', 'matching', 'minkowski', 'rogerstanimoto',
+                             'russellrao', 'seuclidean', 'sokalmichener',
+                             'sokalsneath', 'sqeuclidean', 'yule', 'wminkowski'
+
+**K-D Tree**                 'chebyshev', 'euclidean', 'cityblock', 'manhattan', 'infinity',
+                             'minkowski', 'p', 'l2', 'l1'
+
+**Ball Tree**                'chebyshev', 'sokalmichener', 'canberra', 'haversine',
+                             'rogerstanimoto', 'matching', 'dice', 'euclidean', 'braycurtis',
+                             'russellrao', 'cityblock', 'manhattan', 'infinity', 'jaccard',
+                             'seuclidean', 'sokalsneath', 'kulsinski', 'minkowski',
+                             'mahalanobis', 'p', 'l2', 'hamming', 'l1', 'wminkowski', 'pyfunc'
+========================     =================================================================
+
+A list of valid metrics for the any of the above algorithms can be obtained by using their
+``valid_metric`` attribute. For example, valid metrics for ``KDTree`` can be generated by:
+
+    >>> from sklearn.neighbors import KDTree
+    >>> import numpy as np
+    >>> print(np.sort(KDTree.valid_metrics)) # doctest: +ELLIPSIS
```

Comment:
@raghavrv I don't think we need `numpy` for this doctest, a normal `sorted` would work just fine?